src/Tools/dotnet-monitor/HostBuilder/HostBuilderHelper.cs

```diff
@@ -75,17 +76,6 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                     builder.AddKeyPerFile(path, optional: true, reloadOnChange: true);
                     builder.AddEnvironmentVariables(ToolIdentifiers.StandardPrefix);
 
-                    if (settings.Authentication.KeyAuthenticationMode == KeyAuthenticationMode.TemporaryKey)
```

Comment:
Is there a technical reason why this cannot be moved to right after `FileInfo userFilePath = settings.UserProvidedConfigFilePath;` to fix the intended behavior (`--temp-apikey` always wins), keep the rest of the refactor as-is, and avoid the technical breaking change?

Reply:
Spoke offline; we'll keep the "breaking" change for 8.0. If we backport to `release/7.x`, we'll make an affordance to add back the temporary API key to `config show`.